drivers/video/video_arducam_mega.c

```diff
@@ -0,0 +1,1397 @@
+	case VIDEO_CID_AUTO_EXPOSURE_BIAS:
+		return arducam_mega_set_EV(dev, drv_data->ctrls.ev.val);
+	case VIDEO_CID_SHARPNESS:
+		return arducam_mega_set_sharpness(dev, drv_data->ctrls.sharpness.val);
+	case VIDEO_CID_FOCUS_AUTO:
+		return arducam_mega_set_auto_focus(dev, drv_data->ctrls.focus_auto.val);
+	case VIDEO_CID_COLORFX:
+		return arducam_mega_set_special_effects(
+			dev, drv_data->ctrls.support_special_effects.val);
+	case VIDEO_CID_ARDUCAM_LOWPOWER:
+		return arducam_mega_set_lowpower_enable(dev, drv_data->ctrls.lowpower.val);
+	case VIDEO_CID_ARDUCAM_RESET: {
+		drv_data->ctrls.reset.val = 0;
+		ret = arducam_mega_soft_reset(dev);
+		if (ret < 0) {
+			return ret;
+		}
+		return arducam_mega_check_connection(dev);
+	}
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static const struct video_driver_api arducam_mega_driver_api = {
+	.set_format = arducam_mega_set_fmt,
+	.get_format = arducam_mega_get_fmt,
+	.set_ctrl = arducam_mega_set_ctrl,
+	.get_caps = arducam_mega_get_caps,
+	.set_stream = arducam_mega_stream_start,
+	.flush = arducam_mega_flush,
+	.enqueue = arducam_mega_enqueue,
+	.dequeue = arducam_mega_dequeue,
+};
+
+#define ARDUCAM_MEGA_640_480_LINK_FREQ      120000000
+#define ARDUCAM_MEGA_640_480_LINK_FREQ_ID   0
+#define ARDUCAM_MEGA_1600_1200_LINK_FREQ    240000000
+#define ARDUCAM_MEGA_1600_1200_LINK_FREQ_ID 1
+const int64_t arducam_mega_link_frequency[] = {
+	ARDUCAM_MEGA_640_480_LINK_FREQ,
+	ARDUCAM_MEGA_1600_1200_LINK_FREQ,
+};
+
+static int arducam_mega_init_controls(const struct device *dev)
+{
+	int ret = 0;
+	struct arducam_mega_data *drv_data = dev->data;
+	struct arducam_mega_ctrls *ctrls = &drv_data->ctrls;
+
+	ret = video_init_ctrl(&ctrls->reset, dev, VIDEO_CID_ARDUCAM_RESET,
+			      (struct video_ctrl_range){.min = 0, .max = 1, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->brightness, dev, VIDEO_CID_BRIGHTNESS,
+			      (struct video_ctrl_range){.min = 0, .max = 8, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->contrast, dev, VIDEO_CID_CONTRAST,
+			      (struct video_ctrl_range){.min = 0, .max = 6, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->saturation, dev, VIDEO_CID_SATURATION,
+			      (struct video_ctrl_range){.min = 0, .max = 6, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->ev, dev, VIDEO_CID_AUTO_EXPOSURE_BIAS,
+			      (struct video_ctrl_range){.min = 0, .max = 6, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->whitebal, dev, VIDEO_CID_WHITE_BALANCE_TEMPERATURE,
+			      (struct video_ctrl_range){.min = 0, .max = 4, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->colorfx, dev, VIDEO_CID_COLORFX,
+			      (struct video_ctrl_range){.min = 0, .max = 14, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
```

Comment:
This can be wrapped inside an `if (drv_data->features & MEGA_HAS_COLORFX) { ... }`.